platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/io/SDCardGenerator.xtend

```diff
@@ -141,18 +131,21 @@ class SDCardGenerator extends AbstractSystemResourceGenerator {
 			}
 			if(«fileSeekIndex» >= sdCardFileInfo.fsize)
 			{
-				return EXCEPTION_ENDOFFILEEXCEPTION;
+				sdCardReturn = FR_INVALID_PARAMETER;

```

Comment:
returning `EXCEPTION_ENDOFFILEEXCEPTION` allowed the user to differentiate between end of file and other exceptions. Is it possible to reintroduce this? For example by adding `retcode = EXCEPTION_ENDOFFILEEXCEPTION` here, as well as setting the retcode in other places where errors happened.
Also, shouln't this check happen *before* we seek to the position? It could be as early as directly after `f_stat`, since it's a cheap check that could be true regularly.